### PR TITLE
User can sign up

### PR DESCRIPTION
**Pull request by ViaMarcus**
### In this PR
- Found missing line in Application controller through git history.
- Added feature specs for sign-up form
[PT LINK](https://www.pivotaltracker.com/story/show/172432564)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def conversation
   protected
 
   def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
```
Comment (kermit-klein) [inline app/controllers/application_controller.rb:23]: That looks like a smart solution to the problem we had yesterday

File: spec/features/user_can_sign_up_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    it 'should give an error no name is given' do
+        fill_in "Email", with: "bob@bobert.com"
+        fill_in "Password", with: "qwerty123"
+        fill_in "Password confirmation", with: "qwerty123"
+        click_on "Create"
+        expect(page).to have_content "Name can't be blank"
+    end
+
+    it 'should give an error if passwords dont match' do
```
Comment (PaulineBA) [inline spec/features/user_can_sign_up_spec.rb:29]: doesn't* match

File: spec/features/user_can_sign_up_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+feature 'user can sign up' do
+    before do
+        visit root_path
+        click_on "Sign up"
+    end
+
+    it 'should have a signup form' do
+        expect(page).to have_content "Sign up"
+        expect(page).to have_content "Password"
```
Comment (faraznaeem) [inline spec/features/user_can_sign_up_spec.rb:9]: We avoid adding two expectations in the same `it` block

File: spec/features/user_can_sign_up_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    it 'user can fill form and submit' do
+        fill_in "Name", with: "Bobert"
+        fill_in "Email", with: "bob@bobert.com"
+        fill_in "Password", with: "qwerty123"
+        fill_in "Password confirmation", with: "qwerty123"
+        click_on "Create"
+        expect(page).to have_content "You have signed up successfully"
```
Comment (faraznaeem) [inline spec/features/user_can_sign_up_spec.rb:18]: Split the tests into separate `describe` blocks, and extract the fill in etc to a `before` block as well

File: spec/features/user_can_sign_up_spec.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        end
+    end
+    describe 'user gets informative error messages' do
+        before do
+            visit root_path
+            click_on "Sign up"
+            fill_in "Email", with: "bob@bobert.com"
+            fill_in "Password", with: "qwerty123"
+        end
+        it 'should give an error no name is given' do
```
Comment (faraznaeem) [inline spec/features/user_can_sign_up_spec.rb:28]: Line

File: spec/features/user_can_sign_up_spec.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        end
+
+        it 'should give an error if passwords dont match' do
+            fill_in "Name", with: "Bobby"
+            fill_in "Password confirmation", with: "qwerty124"
+            click_on "Create"
+            expect(page).to have_content "Password confirmation doesn't match Password"
+        end
+
+        it 'does not allow duplicate signups' do
```
Comment (faraznaeem) [inline spec/features/user_can_sign_up_spec.rb:41]: new describe and move to before block

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -15,4 +15,12 @@
   config.use_transactional_fixtures = true
   config.infer_spec_type_from_file_location!
   config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
 end
+
+Shoulda::Matchers.configure do |config|
+	config.integrate do |with|
+		with.test_framework :rspec
```
Comment (faraznaeem) [inline spec/rails_helper.rb:23]: formatting

File: spec/features/user_can_sign_up_spec.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            fill_in "Email", with: "bob@bobert.com"
+            fill_in "Password", with: "qwerty123"
+            fill_in "Password confirmation", with: "qwerty123"
+            click_on "Create"
+            click_on "Logout"
+            click_on "Sign up"
+            fill_in "Name", with: "Bobert"
+            fill_in "Email", with: "bob@bobert.com"
+            fill_in "Password", with: "qwerty123"
+            fill_in "Password confirmation", with: "qwerty123"
```
Comment (faraznaeem) [inline spec/features/user_can_sign_up_spec.rb:54]: Not a blocker, but you could create a user with factory bot, give it the attibutes that need to be checked for and then test if you can sign up with the same name

**Pull request merged**